### DISCUSSION TITLE
import of js files outside the project directory fails

### DISCUSSION THREAD

**Discussion by andrisi** [quasarframework/quasar#15572] [General - Components / Directives / etc]
I'm trying to use a js file with some const values shared between projects, but import fails because the file is not treated as a module. Works find if I copy the file into the project directory, but defeats the purpose. Some config issue I guess.

`import * as c from '../../src/const.js'`

This leads to the below error as the file is not treated as a module. Adding a package.json file does not help.

`Parsing error: The keyword 'export' is reserved`

**Comment by MilosPaunovic** [comment]
How are you exporting data from your `.js` file?

**Reply by andrisi** [reply]
with a bunch of consts like this below, and of course they work ok in the server side nodjs poject:

```
export const textFontentForms = [
    'txt', 
    'htxt', 
    'stxt', 
    'ptxt'
]
```

**Comment by MilosPaunovic** [comment]
Quasar application isn't being run in the `Node` context itself, so you should have some something like this in your `constants` file:

```js
const extensions = [ 'txt', 'jpg', 'png' ];

module.exports = { extensions };
```

And you would use it in you `.vue` file like this:

```js
const { extensions } = require('./constants.js');
```

**Reply by andrisi** [reply]
I can use import in vue files otherwise, its just that for some reason it does not recognise files ouside the project directory as module files. The same files works just fin if I move it inside the Quasar project directory.

**Reply by fprijate** [reply]
Make **symlinks** (junction on Windows)  in all app folders to shared folder.

**Reply by andrisi** [reply]
nice hack, but would not work with git, and I'd rather fix the underlying config issue with vite or something

**Reply by fprijate** [reply]
Then solution is monorepo with  workspace.
